federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -931,31 +896,10 @@ func (s *ServiceController) processServiceUpdate(cachedService *cachedService, s
 // we should retry in that Duration.
 func (s *ServiceController) processServiceDeletion(key string) (error, time.Duration) {
 	glog.V(2).Infof("Process service deletion for %v", key)
-	cachedService, ok := s.serviceCache.get(key)
-	if !ok {
-		return fmt.Errorf("Service %s not in cache even though the watcher thought it was. Ignoring the deletion.", key), doNotRetry
-	}
-	service := cachedService.lastState
-	cachedService.rwlock.Lock()
-	defer cachedService.rwlock.Unlock()
-	s.eventRecorder.Event(service, v1.EventTypeNormal, "DeletingDNSRecord", "Deleting DNS Records")
-	// TODO should we delete dns info here or wait for endpoint changes? prefer here
-	// or we do nothing for service deletion
+	// TODO: Should uncomment this?
+	// We should delete the load balancer when service is deleted
+	// Look at the same method in kube service controller.
 	//err := s.dns.balancer.EnsureLoadBalancerDeleted(service)
```

Comment:
@nikhiljindal by "this" I meant the TODO comment and commented code. What was your "this"?

What loadbalancer do you think we are going to delete here? Also, the commented code is wrong anyway. `s.dns` is an interface that doesn't contain a field called `balancer`. My gut feeling is that, this was a copy-paste from cluster-level service controller and doesn't apply here. 

Please remove it. The comment change is making things worse. And the right fix for that is just removing it.